util/src/main/java/io/kubernetes/client/util/CallGenerator.java

```diff
@@ -13,13 +13,13 @@
  * params.resourceVersion, params.timeoutSeconds, params.watch, null, null); },
  */
 @FunctionalInterface
-public interface CallGenerator {
+public interface CallGenerator<T extends CallGeneratorParams> {
```

Comment:
Why is templating necessary here? It seems that polymorphism would take care of it.

Reply:
PageSharedInformerFactory use PagerCallGeneratorParams ,but ListerWatcher  list method hardcode  CallGeneratorParams